From dad1a4c45dda747b6bb63b1a10daff4e816f5979 Mon Sep 17 00:00:00 2001
From: Karthik M <quic_karm@quicinc.com>
Date: Thu, 8 Dec 2022 18:46:56 +0530
Subject: [PATCH] ath11k : regdb separate binary support removal

lithium architecture no longer needed support for download of
separate regdb.bin. It will be encoded as part of board-2.bin.

Signed-off-by: Karthik M <quic_karm@quicinc.com>
---
 drivers/net/wireless/ath/ath11k/core.c | 12 +++---------
 drivers/net/wireless/ath/ath11k/qmi.c  |  2 +-
 2 files changed, 4 insertions(+), 10 deletions(-)

diff --git a/drivers/net/wireless/ath/ath11k/core.c b/drivers/net/wireless/ath/ath11k/core.c
index 1db2334..49d514d 100644
--- a/drivers/net/wireless/ath/ath11k/core.c
+++ b/drivers/net/wireless/ath/ath11k/core.c
@@ -1554,7 +1554,6 @@ int ath11k_core_fetch_regdb(struct ath11k_base *ab, struct ath11k_board_data *bd
 		return ret;
 	}
 
-	ab->bd_api = 2;
 	ret = ath11k_core_fetch_board_data_api_n(ab, bd, regdbname,
 						 ATH11K_BD_IE_REGDB,
 						 ATH11K_BD_IE_REGDB_NAME,
@@ -1566,19 +1565,14 @@ int ath11k_core_fetch_regdb(struct ath11k_base *ab, struct ath11k_board_data *bd
 						 ATH11K_BD_IE_REGDB,
 						 ATH11K_BD_IE_REGDB_NAME,
 						 ATH11K_BD_IE_REGDB_DATA);
-	if (!ret)
-		goto success;
-
-	ab->bd_api = 1;
-	ret = ath11k_core_fetch_board_data_api_1(ab, bd, ATH11K_REGDB_FILE_NAME);
 	if (ret) {
-		ath11k_dbg(ab, ATH11K_DBG_BOOT, "failed to fetch %s from %s\n",
-			   ATH11K_REGDB_FILE_NAME, ab->hw_params.fw.dir);
+		ath11k_dbg(ab, ATH11K_DBG_BOOT, "failed to fetch %s from board-2.bin\n",
+			   ATH11K_REGDB_FILE_NAME);
 		return ret;
 	}
 
 success:
-	ath11k_dbg(ab, ATH11K_DBG_BOOT, "using board api %d\n", ab->bd_api);
+	ath11k_dbg(ab, ATH11K_DBG_BOOT, "using board-2.bin\n");
 	return 0;
 }
 
diff --git a/drivers/net/wireless/ath/ath11k/qmi.c b/drivers/net/wireless/ath/ath11k/qmi.c
index 64c05d9..6fc1159 100644
--- a/drivers/net/wireless/ath/ath11k/qmi.c
+++ b/drivers/net/wireless/ath/ath11k/qmi.c
@@ -4319,7 +4319,7 @@ static int ath11k_qmi_event_load_bdf(struct ath11k_qmi *qmi)
 		return ret;
 	}
 
-	if (ab->hw_params.supports_regdb)
+	if (ab->hw_params.supports_regdb  && ab->qmi.target.regdb)
 		ath11k_qmi_load_bdf_qmi(ab, true);
 
 	ret = ath11k_qmi_load_bdf_qmi(ab, false);
-- 
2.17.1

